cmd/nodeJsBuild.go

```diff
@@ -0,0 +1,157 @@
+package cmd
+
+import (
+	"github.com/SAP/jenkins-library/pkg/command"
+	"github.com/SAP/jenkins-library/pkg/log"
+	FileUtils "github.com/SAP/jenkins-library/pkg/piperutils"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+	"github.com/bmatcuk/doublestar"
+	"os"
+	"path"
+	"strings"
+)
+
+type nodeJsBuildUtilsInterface interface {
+	fileExists(path string) (bool, error)
+	glob(pattern string) (matches []string, err error)
+	getwd() (dir string, err error)
+	dir(path string) string
+	chdir(dir string) error
+	getExecRunner() execRunner
+}
+
+type nodeJsBuildUtilsBundle struct {
+	fileUtils  FileUtils.Files
+	execRunner *command.Command
+}
+
+func (u *nodeJsBuildUtilsBundle) fileExists(path string) (bool, error) {
+	return u.fileUtils.FileExists(path)
+}
+
+func (u *nodeJsBuildUtilsBundle) glob(pattern string) (matches []string, err error) {
+	return doublestar.Glob(pattern)
+}
+
+func (u *nodeJsBuildUtilsBundle) getwd() (dir string, err error) {
+	return os.Getwd()
+}
+
+func (u *nodeJsBuildUtilsBundle) dir(fileName string) string {
+	return path.Dir(fileName)
+}
+
+func (u *nodeJsBuildUtilsBundle) chdir(dir string) error {
+	return os.Chdir(dir)
+}
+
+func (u *nodeJsBuildUtilsBundle) getExecRunner() execRunner {
+	if u.execRunner == nil {
+		u.execRunner = &command.Command{}
+		u.execRunner.Stdout(log.Entry().Writer())
+		u.execRunner.Stderr(log.Entry().Writer())
+	}
+	return u.execRunner
+}
+
+func nodeJsBuild(config nodeJsBuildOptions, telemetryData *telemetry.CustomData) {
+	utils := nodeJsBuildUtilsBundle{}
+
+	err := runNodeJsBuild(&utils, &config)
```

Comment:
You may have to process the parameter `runScripts`. The Groovy layer has the habit of passing command line options which should be a list as a single string containing multiple options separated by spaces. This didn't matter in Groovy, because the command line is concatenated anyway, and then the shell will process it before handing it to the command. Using the execRunner, you have to do the processing yourself. Maybe it doesn't apply here, just something to be aware of. If it is not clear what I mean, see the change I had to do here: https://github.com/SAP/jenkins-library/pull/1388/files#diff-e7149f65cdaf6eadb7bf5564133dad8fR28-R30